Error while checking build/O2/fullCI for b41b17324e4b60cc6bf17af9c9aa86190f97cfb9 at 2024-08-29 10:37:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13449-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/src/TrackingInterface.cxx:169:11: error: 'rackROFspan' was not declared in this scope; did you mean 'trackROFspan'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13449/b41b17324e4b60cc6bf17af9c9aa86190f97cfb9/build_O2_fullCI/pretty.html).

--- a/Detectors/ITSMFT/ITS/tracking/include/ITStracking/TrackingInterface.h
+++ b/Detectors/ITSMFT/ITS/tracking/include/ITStracking/TrackingInterface.h
@@ -57,8 +57,8 @@ class ITSTrackingInterface
   template <bool isGPU = false>
   void run(framework::ProcessingContext& pc);
 
-  void updateTimeDependentParams(framework::ProcessingContext& pc);
-  void finaliseCCDB(framework::ConcreteDataMatcher& matcher, void* obj);
+  virtual void updateTimeDependentParams(framework::ProcessingContext& pc);
+  virtual void finaliseCCDB(framework::ConcreteDataMatcher& matcher, void* obj);
 
   // Custom
   void setTraitsFromProvider(VertexerTraits*, TrackerTraits*, TimeFrame*);
@@ -70,6 +70,15 @@ class ITSTrackingInterface
     mMode = mode;
   }
 
+  TimeFrame* mTimeFrame = nullptr;
+
+ protected:
+  virtual void loadROF(gsl::span<itsmft::ROFRecord>& trackROFspan,
+                       gsl::span<const itsmft::CompClusterExt> clusters,
+                       gsl::span<const unsigned char>::iterator& pattIt,
+                       const dataformats::MCTruthContainer<MCCompLabel>* mcLabels);
+  void getConfiguration(framework::ProcessingContext& pc);
+
  private:
   bool mIsMC = false;
   bool mRunVertexer = true;
@@ -80,7 +89,6 @@ class ITSTrackingInterface
   const o2::itsmft::TopologyDictionary* mDict = nullptr;
   std::unique_ptr<Tracker> mTracker = nullptr;
   std::unique_ptr<Vertexer> mVertexer = nullptr;
-  TimeFrame* mTimeFrame = nullptr;
   const o2::dataformats::MeanVertexObject* mMeanVertex;
 };
 

--- a/Detectors/ITSMFT/ITS/tracking/src/TrackingInterface.cxx
+++ b/Detectors/ITSMFT/ITS/tracking/src/TrackingInterface.cxx
@@ -166,7 +166,7 @@ void ITSTrackingInterface::run(framework::ProcessingContext& pc)
   gsl::span<const unsigned char>::iterator pattIt = patterns.begin();
 
   gsl::span<itsmft::ROFRecord> trackROFspan(trackROFvec);
-  mTimeFrame->loadROFrameData(trackROFspan, compClusters, pattIt, mDict, labels);
+  loadROF(trackROFspan, compClusters, pattIt, labels);
   pattIt = patterns.begin();
   std::vector<int> savedROF;
   auto logger = [&](std::string s) { LOG(info) << s; };
@@ -343,11 +343,16 @@ void ITSTrackingInterface::updateTimeDependentParams(framework::ProcessingContex
     }
     GeometryTGeo* geom = GeometryTGeo::Instance();
     geom->fillMatrixCache(o2::math_utils::bit2Mask(o2::math_utils::TransformType::T2L, o2::math_utils::TransformType::T2GRot, o2::math_utils::TransformType::T2G));
-    mVertexer->getGlobalConfiguration();
-    mTracker->getGlobalConfiguration();
-    if (mOverrideBeamEstimation) {
-      pc.inputs().get<o2::dataformats::MeanVertexObject*>("meanvtx");
-    }
+    getConfiguration(pc);
+  }
+}
+
+void ITSTrackingInterface::getConfiguration(framework::ProcessingContext& pc)
+{
+  mVertexer->getGlobalConfiguration();
+  mTracker->getGlobalConfiguration();
+  if (mOverrideBeamEstimation) {
+    pc.inputs().get<o2::dataformats::MeanVertexObject*>("meanvtx");
   }
 }
 
@@ -389,6 +394,14 @@ void ITSTrackingInterface::setTraitsFromProvider(VertexerTraits* vertexerTraits,
   mTimeFrame = frame;
   mVertexer->adoptTimeFrame(*mTimeFrame);
   mTracker->adoptTimeFrame(*mTimeFrame);
+}
+
+void ITSTrackingInterface::loadROF(gsl::span<itsmft::ROFRecord>& trackROFspan,
+                                   gsl::span<const itsmft::CompClusterExt> clusters,
+                                   gsl::span<const unsigned char>::iterator& pattIt,
+                                   const dataformats::MCTruthContainer<MCCompLabel>* mcLabels)
+{
+  mTimeFrame->loadROFrameData(trackROFspan, clusters, pattIt, mDict, mcLabels);
 }
 
 template void ITSTrackingInterface::run<true>(framework::ProcessingContext& pc);

